homeassistant/components/zha/device_tracker.py

```diff
@@ -47,11 +50,20 @@
     """Set up the ZHA device trackers."""
     entities = []
     for discovery_info in discovery_infos:
-        entities.append(ZHADeviceScannerEntity(**discovery_info))
+        zha_dev = discovery_info["zha_device"]
+        channels = discovery_info["channels"]
 
-    async_add_entities(entities, update_before_add=True)
+        entity = ZHA_ENTITIES.get_entity(
+            DOMAIN, zha_dev, channels, ZHADeviceScannerEntity
+        )
+        if entity:
+            entities.append(entity(**discovery_info))
+
+    if entities:
+        async_add_entities(entities, update_before_add=True)
```

Comment:
Should we break this out into a helper? seems to be the same 14 or so lines of code in every platform. The only differences are the domain and entity class.

Reply:
I concur, but I'm testing a few approaches and expect this code to go away in subsequent PRs so left it as is with minimal changes.